core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/stopstartus/StopStartUpscaleFlowService.java

```diff
@@ -47,19 +49,32 @@ class StopStartUpscaleFlowService {
     @Inject
     private InstanceMetaDataService instanceMetaDataService;
 
-    void startingInstances(long stackId, String hostGroupName, int nodeCount) {
+    void getRecoveryCandidates(long stackId, String hostGroupName) {
         stackUpdater.updateStackStatus(stackId, DetailedStackStatus.UPSCALE_BY_START_IN_PROGRESS,
                 String.format("Scaling up (stopstart) host  group: %s", hostGroupName));
-        flowMessageService.fireEventAndLog(stackId, UPDATE_IN_PROGRESS.name(), CLUSTER_SCALING_STOPSTART_UPSCALE_INIT, hostGroupName, String.valueOf(nodeCount));
+        flowMessageService.fireEventAndLog(stackId, UPDATE_IN_PROGRESS.name(), CLUSTER_SCALING_STOPSTART_UPSCALE_RECOVERYCANDIDATES, hostGroupName);
+    }
+
+    void startingInstances(long stackId, String hostGroupName, int nodeCount, List<CloudInstance> recoveryCandidates) {
+        if (recoveryCandidates.isEmpty()) {
+            flowMessageService.fireEventAndLog(stackId, UPDATE_IN_PROGRESS.name(), CLUSTER_SCALING_STOPSTART_UPSCALE_INIT, hostGroupName,
```

Comment:
This needs to always be the first line that is logged, with or without recovery... subsequent steps follow after this.

Reply:
For the upscale case - this INIT event is a little broken. What we want the log to be is along the lines of

1. Starting to process upscale request (i.e. one line to indicate the overall operation has started).
2. After this - Identifying recovery candidates, OR Starting N instances via cloud provider.
3. Rest of the flow.

Reply:
Reordered these messages